### PR TITLE
[IMP] project,hr_timesheet,sale_timesheet: generic improvements

### PR DESCRIPTION
**Pull request by kro-odoo**
The purpose of this commit, is to make improvements in Project.

In this commit, following changes are made:

-rename 'Duration' to Hours/Days Spent
-rename hours/days to Hours/Days Spent
-add links to Sales Order, Quotation, Ticket and Invoices linked
 the task in left panel
-indicate email of the assignee
-indicate email of the partner
-indicate 'Draft Invoice' instead of '/' for invoices
-display 'remaning hours on SO' value in red if value is less
 than 0

task-2743512
--
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

File: addons/sale_timesheet/controllers/portal.py [hunk 1]
```diff
@@ -1,19 +1,23 @@
...
 
-from odoo.addons.account.controllers import portal
+from odoo.addons.account.controllers.portal import PortalAccount
 from odoo.addons.hr_timesheet.controllers.portal import TimesheetCustomerPortal
+from odoo.addons.portal.controllers.portal import pager as portal_pager
+from odoo.addons.project.controllers.portal import ProjectCustomerPortal
 
 
-class PortalAccount(portal.PortalAccount):
+class PortalProjectAccount(PortalAccount, ProjectCustomerPortal):
```
Comment (xavierbol) [inline addons/sale_timesheet/controllers/portal.py:17]: Why not directly inheriting to only `ProjectCustomerPortal`? 🤔

File: addons/sale_timesheet/controllers/portal.py [hunk 2]
```diff
@@ -76,16 +107,28 @@ def _task_get_page_view_values(self, task, access_token, **kwargs):
...
 
-        values['invoices_accessible'] = []
-        for invoice in task.sale_order_id.invoice_ids:
-            try:
-                if self._document_check_access('account.move', invoice.id):
-                    values['invoices_accessible'].append(invoice.id)
-            except (AccessError, MissingError):
-                pass
+        if task.sale_order_id.invoice_ids:
+            moves = request.env['account.move'].search([('id', 'in', task.sale_order_id.invoice_ids.ids)])
```
Comment (xavierbol) [inline addons/sale_timesheet/controllers/portal.py:119]: You already have the invoices with `task.sale_order_id.invoice_ids`, so why you did a search? 🤔

File: addons/sale_timesheet/controllers/portal.py [hunk 3]
```diff
@@ -76,16 +107,28 @@ def _task_get_page_view_values(self, task, access_token, **kwargs):
...
+            values['invoices_accessible'] = moves.ids
+            if moves:
+                if len(moves) == 1:
+                    task_invoice_url = f'/my/invoices/{moves.id}'
```
Comment (xavierbol) [inline addons/sale_timesheet/controllers/portal.py:123]: Cannot we use the `get_portal_url` method? 🤔

**Review by xavierbol — COMMENTED** [review]
Thanks for your work, I have made some comments. 🙂

File: addons/sale_timesheet/controllers/portal.py (hunk 2 above)
Comment (kro-odoo) [reply addons/sale_timesheet/controllers/portal.py:119]: There might be a case where the user does not have rights to access all the invoices, in order to get the records which user has rights to we did a search.

File: addons/sale_timesheet/controllers/portal.py (hunk 1 above)
Comment (kro-odoo) [reply addons/sale_timesheet/controllers/portal.py:17]: As this controller is to get the invoices, we need to use _prepare_my_invoices_values method which is in PortalAccount.

Comment (xavierbol) [reply addons/sale_timesheet/controllers/portal.py:17]: Oh indeed, sorry, thanks

File: addons/sale_timesheet/controllers/portal.py (hunk 2 above)
Comment (xavierbol) [inline addons/sale_timesheet/controllers/portal.py:119]: Okay, I see, thanks for the explanation. 🙂

**Review by xavierbol — APPROVED** [review]
Thanks for changes and your work, LGTM. 🙂

**Review by IT-Ideas — APPROVED** [review]
Thanks 🙂 
@robodoo r+

**Pull request closed without merging**